tests/ui/components/filters/filteroptionscomponent.js

```diff
@@ -233,3 +229,267 @@ describe('filter options component', () => {
+      container: '#test-component',
+      options: [
+        {
+          label: 'label1',
+          field: 'field',
+          value: 'val1'
+        },
+        {
+          label: 'label2',
+          field: 'field',
+          value: 'val2',
+          selected: false
+        },
+        {
+          label: 'label3',
+          field: 'field',
+          value: 'val3',
+          selected: true
+        },
+        {
+          label: 'label4',
+          field: 'field',
+          value: 'val4',
+          selected: true
+        }
+      ]
+    };
+  });
+
+  it('has no selected options by default', () => {
+    const config = {
+      ...defaultConfig,
+      control: 'singleoption',
+      options: [
+        {
+          label: 'label1',
+          field: 'field',
+          value: 'val1'
+        },
+        {
+          label: 'label2',
+          field: 'field',
+          value: 'val2'
+        },
+        {
+          label: 'label3',
+          field: 'field',
+          value: 'val3'
+        }
+      ]
+    };
+
+    const component = COMPONENT_MANAGER.create('FilterOptions', config);
+    const options = component.config.options;
+    expect(options).toHaveLength(3);
+    expect(options.filter(o => o.selected)).toHaveLength(0);
+  });
+
+  it('can choose selected options for multioption', () => {
+    const config = {
+      ...defaultConfig,
+      control: 'multioption'
+    };
+
+    const component = COMPONENT_MANAGER.create('FilterOptions', config);
+    const options = component.config.options;
+    expect(options).toHaveLength(4);
+    const selectedOptions = options.filter(o => o.selected);
+    expect(selectedOptions).toHaveLength(2);
+    expect(selectedOptions[0].label).toEqual('label3');
+    expect(selectedOptions[1].label).toEqual('label4');
+  });
+
+  it('prioritizes previous options over selected config for multioption', () => {
+    const config = {
+      ...defaultConfig,
+      name: 'test-previous-options',
+      control: 'multioption'
+    };
+
+    const component = COMPONENT_MANAGER.create('FilterOptions', config);
+    const options = component.config.options;
+    expect(options).toHaveLength(4);
+    const selectedOptions = options.filter(o => o.selected);
+    expect(selectedOptions).toHaveLength(2);
+    expect(selectedOptions[0].label).toEqual('label1');
+    expect(selectedOptions[1].label).toEqual('label2');
+  });
+
+  it('can choose selected option for singleoption', () => {
+    const config = {
+      ...defaultConfig,
+      control: 'singleoption'
+    };
+
+    const component = COMPONENT_MANAGER.create('FilterOptions', config);
+    const options = component.config.options;
+    expect(options).toHaveLength(4);
+    const selectedOptions = options.filter(o => o.selected);
+    expect(selectedOptions).toHaveLength(1);
+    expect(selectedOptions[0].label).toEqual('label3');
+  });
+
+  it('prioritizes previous options over selected config for singleoption', () => {
```

Comment:
'prioritizes previously selected option over selected option in config for singleoption'